src/client/src/orangehrmBuzzPlugin/components/PostFilters.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <div class="orangehrm-post-filters">
+    <oxd-button
+      :label="isActiveRecent || !isMobile ? 'Most Recent Posts' : ''"
+      icon-name="clock-history"
+      :display-type="isActiveRecent ? 'label-warn' : 'text'"
+      :class="
+        isActiveRecent || !isMobile ? 'orangehrm-post-filters-button' : ''
+      "
+      @click="filterMostRecent"
+    />
+    <oxd-button
+      :label="isActiveLike || !isMobile ? 'Most Liked Posts' : ''"
+      icon-name="heart-fill"
+      :display-type="isActiveLike ? 'label-warn' : 'text'"
+      :class="
+        isActiveLike || !isMobile
+          ? 'orangehrm-post-filters-button-like'
+          : 'orangehrm-post-filters-button-like-mobile'
+      "
+      @click="filterMostLiked"
+    />
+    <oxd-button
+      :label="isActiveComment || !isMobile ? 'Most Commented Posts' : ''"
+      icon-name="chat-dots-fill"
+      :display-type="isActiveComment ? 'label-warn' : 'text'"
+      :class="
+        isActiveComment || !isMobile ? 'orangehrm-post-filters-button' : ''
+      "
+      @click="filterMostCommented"
+    />
+  </div>
+</template>
+<script>
+export default {
+  name: 'PostFIlters',
+
+  props: {
+    isMobile: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  data() {
+    return {
+      isActiveRecent: true,
+      isActiveLike: false,
+      isActiveComment: false,
+    };
+  },
+
+  methods: {
+    filterMostRecent() {
+      this.isActiveRecent = true;
+      this.isActiveLike = false;
+      this.isActiveComment = false;
+      // TODO - add filters
```

Comment:
emit this action to parent and handle the filter at buzz newsfeed level
https://github.com/orangehrm/orangehrm/blob/5.x-buzz/src/client/src/orangehrmBuzzPlugin/components/NewsFeed.vue#L67